text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+- Feature Name: `partial_types`
+- Start Date: 2023-04-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+
+# Summary
+[summary]: #summary
+
+Partial types proposal is a generalization on "partial borrowing"-like proposals (more correct name is "partial not borrowing" or "qualified borrowing" since Rust allows partial borrowing already).
+
+This proposal is a universal roadmap "how to do partial not consumption (including partial not borrowing) right", and not under the hood of the Rust compiler.
+
+Partial Types is a **minimal** and full extension to the Rust Type System, which allows to safe control easily partial parameters and all kinds of partial not consumption.
+
+Advantages: maximum type safety, maximum type control guarantee, no ambiguities, flexibility, usability and universality.
+
+
+# Motivation
+[motivation]: #motivation
+
+Safe, Flexible controllable partial parameters for functions and partial not consumption (including partial not borrowing) are highly needed and this feature unlock huge amount of possibilities.
+
+Partial borrowing is already possible in Rust, as partial referencing and partial moves.
+
+But partial parameters are forbidden now, as qualified consumption: partial not borrowing, partial not referencing, partial not moving and partial initializing.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+_**Note**: I didn't comment type explanations (as is needed for compiler) for colorizing purposes only._
+
+_**Note**: I use symbol `~` in code as a synonym of "equivalent" word only. It is not a Rust operator!_
+
+
+## Partial types by type Integrity
+
+```rust
+// case (A1)
+let foo : i16 = 0;
+```
+
+What is a **full type** of `foo`? Is it the `i16`? No, the `i16` is a sub-full_type in Rust type-system.
+```rust
+// case (A2)
+<variable>  : <full_type>;
+<variable>  : <type_clarification> <type>;
+<variable>  : <sharing> <'lifetime> <mutability> <type>;
+```
```

Comment:
This feels… extremely wrong, to me. Not only are you explaining with an example that isn't very useful to the guide-level (`i16` is not why we're using this feature), but also, you're not borrowing it at all.

Mutability and lifetime are only part of a type when you're borrowing; otherwise, they're part of the binding, not the type. This seems to dangerously conflate this for the case of values, which ends up making incorrect assumptions.

For example: the lifetime of this variable decidedly *isn't* static, since it's labelled as `let` and not `static`. If you borrow this value, you *cannot* have a static lifetime, since it's dropped at the end of scope.

Getting to this point alone makes me question whether this system makes sense in the first place, but I'll continue reading and maybe I'll be convinced otherwise.

Reply:
Ok, I just cut unnecessary explanations: it looks they confuse people, but not help them.
Thanks for pointing to this